src/libstd/io/impls.rs

```diff
@@ -76,6 +112,15 @@ impl<'a> Write for &'a mut [u8] {
         *self = b;
         Ok(amt)
     }
+
+    fn write_all(&mut self, data: &[u8]) -> io::Result<()> {
+        if try!(self.write(data)) == data.len() {
+            Ok(())
+        } else {
+            Err(Error::new(ErrorKind::WriteZero, "failed to write whole buffer", None))
```

Comment:
This is indeed what `WriteZero` is intended for, but I think `PartialWrite` is actually a better name for it! cc @alexcrichton